### PR TITLE
Fix TextFields, added support for delegate, make Demo.app an Agent, general cleanup, update README

### PR DESCRIPTION
**Pull request by ShadowLightz**
I think you know the issue with NSPopover in the Status Bar, I researched in the web about it and finally solved this problem. I also define a protocol for delegating, I think this can be useful. To simplify some tasks I also add `togglePopover`and `togglePopoverAnimated:`.
Demo.app is now an agent. When I make the Popover FirstResponder, the MainMenu appeared and in addition there were some troubles with fullscreen apps. This change fixes this.

In the moment there is a problem when you open an additional Popover in the Status-Popover. To resolve this, you need to close this before you close the Status-Popover. To do this, the delegate method`popupWillClose`can be used. I added a note in the README.

For better overview I added the necessary NSWindow-category to AXStatusItemPop.h und .m, so these two files are the only needed in future, too.

The other pull request isn't needed anymore. This also fixes the trouble with the Popup-Project.

I updated the Demo-Project (there are two new buttons to toggle the popover and the popover now contains a NSTextField).

### CODE REVIEW

File: AXStatusItemPopup/AXStatusItemPopup.m [hunk 1]
```diff
@@ -121,9 +159,100 @@ - (void)setAlternateImage:(NSImage *)image
...
+    [self togglePopoverAnimated:self.isAnimated];
+}
+
+- (void) togglePopoverAnimated:(BOOL)animated
+{
+    if (self.isActive)
+    {
+        [self hidePopover];
+    } else {
+        [self showPopoverAnimated:self.isAnimated];
```
Comment (ArthurMaroulier) [inline AXStatusItemPopup/AXStatusItemPopup.m]: Does it should be [self showPopoverAnimated:animated]; here, instead of self.isAnimated?

Comment (ShadowLightz) [inline AXStatusItemPopup/AXStatusItemPopup.m]: Of course your right. I'll change it and update the pull request.

**Comment by jeff-h** [comment]
I can confirm this commit is great. Would be good to get it rolled in.

**Comment by ShadowLightz** [comment]
Thanks :)
In the meantime you can use it from my github, I added there some further improvements cocerning focus handling (but is’t still not completely bugfree). As soon I have time, I will take a look into it again and make a setup for CocoaPods

**Comment by aschuch** [comment]
@ShadowLightz,

thanks for your effort, the changes look great.
I would be very happy to merge your changes into the main project.

However, I noticed that some parts of the pull request have a different coding style compared to the original project. Here are some examples:
- Leave curly brackets inline with other code, e.g. `if (self) {`
- No  space between return type and method name, e.g. `- (void)popupDidClose;` not `- (void) popupDidClose;`
- It should not be necessary to change the seperator comments to `#` or `*`

Could you make these changes to match the style of the rest of the project?

**Comment by ShadowLightz** [comment]
As stated, I'm busy in the moment. You're right, it would have been better to keep the coding style in mind, I just added the changes as part of my project, later I thought about submitting a pull request. 
As soon I have time (probably in one month or two) I will take a look again into this and decide whether I will rework this and update the pull.

**Comment by aschuch** [comment]
That would be great!
Thanks again for your time to contribute.

**Comment by sotirisncl** [comment]
With this implementation the 
- (void)rightMouseDown:(NSEvent *)theEvent
  {
   ...
  }

doesn't seem to fire on osx 10.7 and 10.8 any ideas?
Strange enough on 10.9 it seems to be working just fine.

The problem seems to be the imageview added programatically to the statusitem.
[self addSubview imageview]

It seems that you have to subclass the imageview and override hitTest method
as described here 
http://stackoverflow.com/questions/18104039/nsview-subview-right-click-not-working-over-nsimageview

**Comment by ShadowLightz** [comment]
Are you still interested in this? I saw that @joeljfischer made some changes, but you didn't merged it yet. I now would have time for it, but as this work is so stupid I want to be sure it's worth it.
The problems with mission control as mentioned in Issue #13 and the crash in #16 are resolved in my current code. The window behavior is still not perfect, but much better and really usable.

**Comment by aschuch** [comment]
@ShadowLightz still interested in it. Have been away for a while, but might have some time on the weekend to look into all open issues.
Thanks for your contributions :+1:

**Comment by ShadowLightz** [comment]
I close this now, I made a new one with the style changes. Look at PR #21.

**Comment by jeff-h** [comment]
Thanks @ShadowLightz !

**Comment by ShadowLightz** [comment]
You're welcome!

**Pull request closed without merging**